### PR TITLE
Sphere and Plane Rendering

### PR DESCRIPTION
**Pull request by gabriel-gardner**

### CODE REVIEW

File: src/entity/mod.rs [hunk 1]
```diff
@@ -32,7 +47,8 @@ impl Entity {
     }
 }
 
-impl BaseEntity for Entity {
+impl BaseEntity for  Entity {
```
Comment (tanaythan) [inline src/entity/mod.rs]: This spacing change is unnecessary.

File: src/entity/plane.rs [hunk 2]
```diff
@@ -51,7 +64,15 @@ impl Plane {
             ),
         }
     }
+
+    pub fn render(&self) {
+        match self.mesh {
+            Some(ref mesh) => mesh.set_position([self.position.x, self.position.y, self.position.z]),
+            None => println!("Can't render w/o graphics!"),
```
Comment (tanaythan) [inline src/entity/plane.rs]: Should we raise an Error here instead?

File: src/main.rs [hunk 3]
```diff
@@ -39,46 +52,20 @@ fn main() {
...
-    let camera = window.factory.orthographic_camera(center, yextent, zrange);
 
-    while window.update() {
-        window.render(&camera);
+    while i <= MAX_STEPS && is_open {
+        println!("-----------------STEP {:?}-----------------", i);
+        state.step(DT);
+        is_open = render(&mut window, &camera, &state.all_entities());
+        i += 1;
+        thread::sleep(time::Duration::from_millis(500));
```
Comment (tanaythan) [inline src/main.rs]: 500 ms is a little high, do we want to lower this?

**Review by tanaythan — COMMENTED** [review]
Left some comments/questions.

File: src/entity/mod.rs (hunk 1 above)
Comment (Krajiyah) [reply src/entity/mod.rs]: fixed

File: src/entity/plane.rs (hunk 2 above)
Comment (Krajiyah) [reply src/entity/plane.rs]: panic at the disco!

File: src/main.rs (hunk 3 above)
Comment (Krajiyah) [reply src/main.rs]: changed to `50`

**Comment by Krajiyah** [comment]
resolved comments and questions

**Review by tanaythan — APPROVED** [review]

**Pull request merged**